libraries/provider_database_mysql_user.rb

```diff
@@ -116,7 +110,7 @@ def ishash?
             test_sql += " AND Db='#{new_resource.database_name}'" if new_resource.database_name
             test_sql_results = test_client.query test_sql
 
-            incorrect_privs = true if test_sql_results.size.zero?
+            incorrect_privs = true if test_sql_results.size == 0 # rubocop:disable Style/ZeroLengthPredicate
```

Comment:
This actually needs to stay this way. I've run into problems with this not working using this method.